backend/ranking.js

```diff
@@ -3,10 +3,184 @@ import { Follow } from "./models/follow.js";
+            }),
+            Like.findAll({
+                where: { UserId: user.id },
+                include: [{ model: Stock }]
+            }),
+            Dislike.findAll({
+                where: { UserId: user.id },
+                include: [{ model: Stock }]
+            }),
+            Friend.findAll({
+                where: { UserId1: user.id },
+                include: [
+                    { model: User, as: "user1" },
+                    { model: User, as: "user2" }
+                ]
+            })
+        ]);
+
+        if (stocksInDatabase.length < 10) {
+            return {
+                status: 422,
+                error: "You need at least 10 stocks in the database to access ranking."
+            };
+        }
+
+        const followsMap = follows.reduce((accum, current) => {
+            accum[current.Stock.ticker] = true;
+            return accum;
+        }, {});
+
+        const likesMap = likes.reduce((accum, current) => {
+            accum[current.Stock.ticker] = true;
+            return accum;
+        }, {});
+
+        const dislikesMap = dislikes.reduce((accum, current) => {
+            accum[current.Stock.ticker] = true;
+            return accum;
+        }, {});
+
+        const friendIDs = connections.map((connection) => connection.user2.id);
+
+        const stocksThatFriendsFollow = await Promise.all(
+            friendIDs.map(async (friendID) => {
+                const friendFollows = await Follow.findAll({
+                    where: { UserId: friendID },
+                    include: [{ model: Stock }]
+                });
+                const stocksTheyFollow = friendFollows.map((friendFollow) => {
+                    return friendFollow.Stock.dataValues;
+                });
+                return stocksTheyFollow;
+            })
+        );
+
+        const stocksThatFriendsLike = await Promise.all(
+            friendIDs.map(async (friendID) => {
+                const friendLikes = await Like.findAll({
+                    where: { UserId: friendID },
+                    include: [{ model: Stock }]
+                });
+                const stocksTheyLike = friendLikes.map((friendLike) => {
+                    return friendLike.Stock.dataValues;
+                });
+                return stocksTheyLike;
+            })
+        );
+
+        const stocksThatFriendsDislike = await Promise.all(
+            friendIDs.map(async (friendID) => {
+                const friendDislikes = await Dislike.findAll({
+                    where: { UserId: friendID },
+                    include: [{ model: Stock }]
+                });
+                const stocksTheyDislike = friendDislikes.map((friendDislike) => {
+                    return friendDislike.Stock.dataValues;
+                });
+                return stocksTheyDislike;
+            })
+        );
+
+        const friendsFollowsMap = {};
+
+        stocksThatFriendsFollow.forEach((friendStocks) => {
```

Comment:
Nit: You have an object you accumulate to and a forEach bellow. That can always be rewritten with a reduce. It takes time to get used to, but it ends up being easier to read eventually.